kevm-pyk/src/kevm_pyk/__main__.py

```diff
@@ -190,7 +190,8 @@ def update_digest(self, digest_file: Path | None) -> None:
         if 'claims' not in digest_dict:
             digest_dict['claims'] = {}
         digest_dict['claims'][self.claim.label] = self.digest
-        digest_file.write_text(json.dumps(digest_dict, indent=4))
+        with SoftFileLock(f'{digest_file}.lock'):
```

Comment:
Reading the file should be also in the critical section to avoid a race condition.